source/indigo_to_melodic.rst

```diff
@@ -0,0 +1,186 @@
+ROS Melodic + Ubuntu 18.04
+==========================
+
+Upgrading Your Robot to ROS Melodic + Ubuntu 18.04
+--------------------------------------------------
+.. WARNING::
+   Read this document in full to ensure you understand the procedures.  It is
+   not straightforward to go back to ROS Indigo/Ubuntu 14.04 after doing this.
+   Ensure your colleagues are on board with doing this upgrade.
+
+This document is a procedure for replacing the contents of your robot's SSD
+with an Ubuntu 18.04 install and ROS Melodic.
+
+Before Upgrade
+++++++++++++++
+
+Back up files from the robot!  There are a few categories of files to back up:
+
+#. Calibration and other robot-specific files. By convention, these are
+   all in ``/etc/ros/[indigo|melodic]/``
+#. Files relating to your research work
+#. Network hardware configuration (for troubleshooting)
```

Comment:
I'll suggest grabbing a record of what's in /opt/ros/indigo. Any suggestion on how to do that or is `dpkg -l | grep ros-indigo > installed_indigo_packages.txt` good enough?

Reply:
That, and simply `tar` the directory...

Because only the latest version is available from the package manager, so if they've installed their own packages in `/opt/ros/*` or happen to be depending on a very specific version which might not be available they'd have it... but of course it's not easy to get back to the state they were in so that's why I wouldn't recommend it.

Reply:
I hadn't realized how small the contents of that directory typically are. I'll make that update.

Reply:
Yeh it's basically just some binarys and header files, nothing too large.

The only thing I would hate to see is if some student who's going to be presenting a thesis soon and is off working on the writing part, then goes back to the lab after the writing is done to show it / grav a video for the defense... and the were depending on whatever version was in there.
They could flash it back to trusty/indigo and un-tar the directory, clone their workspace and no need to worry about upgrading/updating to whatever is now the latest in the indigo or melodic repos